### PR TITLE
Fix bug 1139508: Remove the /media/redesign folder

### PR DESCRIPTION
**Pull request by openjck**
The name redesign is misleading, as we've been using this design for
more than a year. This commit moves all assets from /media/redesign up
into the general-purpose /media folder.

### CODE REVIEW

**Comment by openjck** [comment]
The more we can test this, the better. :smile:

**Comment by stephaniehobson** [comment]
Needs a rebase already 0.0

**Comment by openjck** [comment]
@darkwing: This moves the `promote.js` file. Were third-party services using that file at one point? I notice it just prints to the console now, telling people to download `promote.js` instead. Do we want to continue to print that guidance to the console when people try to access `promote.js` from its old address?

**Comment by openjck** [comment]
Rebased! :fireworks:

File: media/css/demowrap.css [hunk 1]
```diff
@@ -1,68 +1,49 @@
...
-.paging { float: right; text-align: right; padding-right: 35px; }
-.paging ul { margin: 0; list-style: none; }
-.paging li { display: inline; }
-.paging a { float: left; height: 1.5em; padding: .5em 15px .5em; border-left: 1px solid #333; background-image: url("../img/nav-arrows.png"); background-repeat: no-repeat; }
-.paging a:hover, .paging a:focus, .paging a:active { background-color: #333; }
-.paging li.prev a { padding-left: 25px; background-position: 8px -791px; }
-.paging li.next a { padding-right: 25px; background-position: 88% -691px; }
-
-#demoframe { background: #fff; border: 0; margin: 2.5em 0 0; padding: 0; width: 100%; height: 100%; }
+header,hgroup,nav,section,article,aside,footer,iframe{display:block}
```
Comment (stephaniehobson) [inline media/css/demowrap.css]: I like how you converted the source for this and other demo studios CSS into stylus files :)

Comment (openjck) [inline media/css/demowrap.css]: Thanks. It made the most sense. I didn't want us to have to look in two places for stylesheets.

That reminds me, actually, I need to manually remove `demowrap.css` and others from the repository. They already have history, so gitignore won't be enough. Updating...

**Comment by stephaniehobson** [comment]
We need to make corresponding edits to https://github.com/mozilla/mdn-sphinx-theme/blob/master/mdn_theme/mdn/layout.html Is this a separate PR request or should some automatic process be updating it?

**Comment by openjck** [comment]
Great catch. Yeah, we should do that. I'll open a corresponding pull request and then maybe we can coordinate to merge that the same time we push this to prod.

**Comment by openjck** [comment]
Just did a quick rebase. None of the files that were touched in this pull request have been changed, but the branch now includes the updated maintenance page.

**Comment by stephaniehobson** [comment]
This is looking great to me. Very through :) Nice to see some of those unused assets going away!

The only blocker is changes to the Sphinx theme which @openjck is already working on.

**Comment by openjck** [comment]
Corresponding pull request here:

https://github.com/mozilla/mdn-sphinx-theme/pull/4

**Comment by openjck** [comment]
In terms of moving forward, it might be easier to merge this pull request first, push it to production, and then do a quick review/merge of the corresponding Sphinx changes after that. It would be hard to really do a true test of the Sphinx changes without first pushing this to prod.

**Comment by darkwing** [comment]
I took a brief look at this and I'm not seeing any 404s.  What we may want to consider or ask ourselves is if we think there are any external resources "hotlinking" anything in the "redesign/" folder.  If so, do we add a .htaccess redirect for said folder?  Something for all of us to consider.

**Comment by openjck** [comment]
Works for me. I'll add a redirect from `media/redesign/*` to `/media/[match]` which should cover all the bases.

**Comment by stephaniehobson** [comment]
The redirects would fix the need for us to deploy the sphinx changes simultaneously :)

**Comment by openjck** [comment]
Hey, that's true. Awesome!

**Comment by groovecoder** [comment]
cheating

**Comment by openjck** [comment]
Updated with local redirects. I'll need to coordinate with IT to set up redirects on our servers, but we should merge and push this first.

**Comment by openjck** [comment]
Updated :smile:

We now have a permanent redirect locally. I'll coordinate with WebOps to add the redirect on our servers, too. In the meantime, I added a temporary symlink from `media/redesign` to `media` so that the old addresses will continue to work until WebOps adds the redirect.

As a result, we can merge and push this any time without breaking hotlinks or our Sphinx theme.

**Pull request merged**